### ISSUE TITLE
User information is not displayed correctly when viewing muted users in advanced webUI

### ISSUE THREAD

**Issue by naaaaaaaaaaaf** [mastodon/mastodon#27130]
### Steps to reproduce the problem

1. Enable advanced WebUI in the settings page.
2. Click the hamburger menu above the post field and select "Muted Users".
3. The list of users on mute looks like the attached image.
![スクリーンショット 2023-09-25 215045](https://github.com/mastodon/mastodon/assets/28683404/c941c2c3-87e2-4baa-9449-9a7e8b5ced90)

### Expected behaviour

Usernames and icons are displayed unobstructed by other elements

### Actual behaviour

Usernames and icons are not displayed properly because they are obstructed by other elements

### Mastodon instance

mastodon.social

### Mastodon version

v4.2.0

### Browser name and version

Google Chrome 116.0.5845.188

### Operating system

Windows 10 Pro 22H2

**Comment by leevic31** [comment]
I would like to make an attempt to fix this bug

**Comment by splacat** [comment]
Happen the same to me with Safari and Firefox, Mastodon server versions v4.2.1 and v4.3.0-alpha.0

https://github.com/mastodon/mastodon/assets/64102663/2137580f-eca6-4e3e-8b35-4ba226000808

**Comment by ireddick** [comment]
Hi, I've been taking a look at this.

## Overview

The general problem is that the columns in the advanced interface are significantly narrower than the central pane in the standard interface, yet the account details + action buttons always use a horizontal layout.

I think that in this situation the account details + action buttons ought to use a vertical layout, like this:

<img width="346" alt="advanced_mutes_vertical" src="https://github.com/mastodon/mastodon/assets/521563/b1fe5f3f-ab49-4db8-9eca-8c296f4b3a7a">

Additionally, this problem extends beyond the 'Muted users' view - it happens everywhere the account details + action buttons are displayed, eg. profiles returned by searches:

<img width="304" alt="advanced_search_results_current" src="https://github.com/mastodon/mastodon/assets/521563/06fd2009-d1ca-428d-94db-65c02dff9d51">

Because of this, I'd suggest that a vertical layout be used everywhere for account details + action buttons in the advanced interface.

There's also a general issue for small screen sizes, where again the problem is that the account details + action buttons are using a horizontal layout without enough space.
eg. on an iPhone 8, profile search results looks like this:

<img width="247" alt="iphone8_standard_search_results_current" src="https://github.com/mastodon/mastodon/assets/521563/93a98d77-81c0-4afb-a536-793efb905523">

## Potential fixes

Code-wise, all instances of account details + action buttons in the UI use the same Account component, which currently uses a horizontal flex style defined in the CSS class 'account__wrapper'.

Some ideas:

a) Wrap to a vertical layout anywhere if needed
(ie. use 'flex-wrap: wrap').
This is simple, and fixes all 'too narrow' problems for this everywhere, while preserving a horizontal presentation if possible. A problem here is that it can create an inconsistent UI, as whether the buttons appear on the bottom depends on the length of an each account's details.
Here are a couple of examples of how this would also change the standard interface:

<img width="595" alt="standard_search_results_wrapped" src="https://github.com/mastodon/mastodon/assets/521563/fea4e2f5-b778-410b-8762-467976df1779">

<img width="598" alt="standard_search_results_wrapped_uneven" src="https://github.com/mastodon/mastodon/assets/521563/e77500d7-a508-4e33-9483-66e6995ffa3a">

b) Use a fixed horizontal or vertical layout based on the width of the parent container
(ie. use a container query and add 'flex-direction: column' to the Account component style if it's in a narrow container).
This fixes the issue everywhere in the advanced interface, everywhere for small screens, and preserves the current horizontal behaviour for larger screens using the standard interface (as long as the container query targets a width smaller than the main panel).

c) Use a fixed vertical layout in the advanced interface only
(ie. add 'flex-direction: column' to the Account component style
only if the advanced interface is active).
NOTE: this would not fix the general issue for small screens.

d) Use a fixed vertical layout everywhere
(ie. use 'flex-direction: column').
This is simple, and fixes all 'too narrow' problems for this everywhere, at the cost of changing the UI throughout the whole app.
An example of how this would also change the standard interface search results:

<img width="595" alt="standard_search_results_vertical" src="https://github.com/mastodon/mastodon/assets/521563/a3e13bd6-9d82-4ca6-baae-805bfa0c222e">

There are certainly other approaches - I've focused on smallest code changes here, as it's the first time I've looked at the codebase.

**Comment by ireddick** [comment]
I guess there are two things to decide here:

1) What the UI should look like

  eg. use a vertical layout, or something else (kebab menu?)

2) Scope of the fix

  eg. just the muted users view in the advanced interface, or everywhere in the advanced
  interface (including non-muted accounts), etc.

See my previous comment for some options the UI change and scope.

**Comment by ke352802081770314** [comment]
I just noticed the same for normal layout, when UI language is german. The localization has decided on very long button labels, and they press the user details into invisibility. I include a screenshot (I tried - it keeps disappearing). 
I experimented with the CSS and adding
`.account .account__display-name { min-width: 30%;`
will keep showing me meaningful detail of the account. I would find a multi-line layout more clunky and visually confusing, and can live with not seeing the buttons in their full width.
